lib/rules/no-redeclare.js

```diff
@@ -33,72 +45,128 @@ module.exports = {
 
     create(context) {
         const options = {
-            builtinGlobals: context.options[0] && context.options[0].builtinGlobals
+            builtinGlobals: Boolean(
+                context.options.length === 0 ||
+                context.options[0].builtinGlobals !== false
```

Comment:
```suggestion
                context.options[0].builtinGlobals
```

Since `default` is used, I think this will always be a boolean.